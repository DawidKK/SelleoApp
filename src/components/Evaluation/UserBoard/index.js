import React, { Component } from 'react';
import Can from '../../Can';

import { roles } from '../../../rbac-roles';
export class UserBoard extends Component {
  state = {
    role: 'user'
  };

  handleChange = e => this.setState({ role: e.target.value})

  render() {
    const { role } = this.state;

    return (
      <div>
        <h1>Pretend this is Admin board, no kidding</h1>
        <h2>You can be anyone you ever dream!</h2>
        {roles.map(role => (
          <label for={role.role}>
            <input id={role.role} onChange={this.handleChange} type="radio" name="role.role" value={role.role} />
            Do you want be {role.role}?
          </label>
        ))}
        <h3>Who can get access to this page?</h3>
        <Can
          role={role}
          perform="admin-page:visit"
          yes={() => <p>Everyone can get access to this place</p>}
        />
        <h3>What can you do on this page?</h3>
        <Can
          role={role}
          perform="admin-page:edit"
          yes={() => (
            <>
              <p>Feel free to change something</p>
              <button onClick={() => alert('You are boss here')}>Check this out!</button>
            </>
          )}
          no={() => <p>Hold your horses, you are just poor user, don't touch anything</p>}
        />
      </div>
    );
  }
}

export default UserBoard;
