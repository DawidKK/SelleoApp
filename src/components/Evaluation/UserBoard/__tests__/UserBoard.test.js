import React from 'react';
import Mockdate from 'mockdate'
import moment from 'moment'
import { shallow } from 'enzyme';
import UserBoard from '..';

describe('MyComponent', () => {
  const res = {
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz',
    address: {
      street: 'Kulas Light',
      suite: 'Apt. 556',
      city: 'Gwenborough',
      zipcode: '92998-3874',
      geo: {
        lat: '-37.3159',
        lng: '81.1496'
      }
    },
    phone: '1-770-736-8031 x56442',
    website: 'hildegard.org',
    company: {
      name: 'Romaguera-Crona',
      catchPhrase: 'Multi-layered client-server neural-net',
      bs: 'harness real-time e-markets'
    }
  }

  const defaultProps = {
    today: Mockdate.set(moment('11-19-2019'))
  }

  it('renders', () => {
    const wrapper = shallow(<UserBoard {...defaultProps} />)
    wrapper.instance().fetchUser = Promise.resolve(res)

    return wrapper.instance().fetchUser.then(response => {
      expect(response).toEqual(res)
    })
  });

  it('displays proper date', () => {
    const wrapper = shallow(<UserBoard {...defaultProps} />)
    expect(wrapper).toMatchSnapshot()
  })
});
