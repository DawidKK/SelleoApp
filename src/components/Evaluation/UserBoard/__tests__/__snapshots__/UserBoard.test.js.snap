// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent displays proper date 1`] = `
<div>
  <h2>
    Today is 
  </h2>
</div>
`;
